pageserver/src/tenant/timeline.rs

```diff
@@ -2709,28 +2713,45 @@ impl Timeline {
                         assert!(!*expect_initdb_optimization, "expected initdb optimization");
                     }
                 }
-                // normal case, write out a L0 delta layer file.
-                let (delta_path, metadata) = self.create_delta_layer(&frozen_layer).await?;
-                HashMap::from([(delta_path, metadata)])
+                // Normal case, write out a L0 delta layer file.
+                // For delta layers, `create_delta_layer` will not modify the layer map.
```

Comment:
> For delta layers, `create_delta_layer` will not modify the layer map.

Does the `create_delta_layer` handle other layers aside from delta?

If not:

```suggestion
                // `create_delta_layer` will not modify the layer map.
```